src/idiff/idiff.cpp

```diff
@@ -169,6 +169,24 @@ print_subimage(ImageBuf& img0, int subimage, int miplevel)
 }
 
 
+// Append the filename from "first" when "second" is a directory.
+// "second" is an output variable and modified in-place.
+inline void
+add_filename_to_directory(const std::string& first, std::string& second)
+{
+    if (Filesystem::is_directory(second)) {
+        char last_byte = second.at(second.size() - 1);
+        if (last_byte != '/' && last_byte != '\\') {
+#if defined(_MSC_VER)
+            second += '\\';
+#else
+            second += '/';
+#endif
+        }
+        second += Filesystem::filename(first);
+    }
+}
+
```

Comment:
The `Filesystem` class didn't already provide this functionality so adding this to idiff seemed like the right place to keep this for now. If other code ends up needing to build paths then moving this to the Filesystem class might be worth it at that point in time. It didn't seem worth extending its public API just for this one small tweak, though.